L4 resources

Stepanav opened the pull request:
наименования переменных привел в соответствие с принятым стилем:
1. исправил на snake_case
2. исправил названия некоторых элементов на более информативные
3. добвил имя типа в конце id
4. создал ресурсы ко всему, к чему смог
5. создал горизонтальный экран

![Screenshot_20211230_010753](https://user-images.githubusercontent.com/92550022/147707060-fd183ecd-fd2b-42b2-a601-182d9783e1df.png)

![Screenshot_20220103_015912](https://user-images.githubusercontent.com/92550022/147891691-5220cbdc-0bde-4778-b4d0-1bb1892a809b.png)

Stepanav committed:
add grid layout keypad

Stepanav committed:
change to table layout keypad

Stepanav committed:
create second Activity

Stepanav committed:
add id each button

Stepanav committed:
1. переделал в constraintlayout
2. назначил кнопки и присвоил им значения
3. реализовал операции
4. создал кнопку открытия второй activity
5. реализовал сохранение данных при повороте экрана

Stepanav committed:
1. переделал в constraintlayout
2. назначил кнопки и присвоил им значения
3. реализовал операции
4. создал кнопку открытия второй activity
5. реализовал сохранение данных при повороте экрана
6. добавил возможность внесения отрицательных чисел

DumDumbIchGB commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -29,40 +29,72 @@
         android:id="@+id/numberField"
```
calculation_result_text_view

DumDumbIchGB commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -29,40 +29,72 @@
 
     <LinearLayout
-        android:id="@+id/firstButtonPanel"
+        android:id="@+id/CEC"
```
я не понимаю, что вы имели ввиду
а после каникул и вы сами не вспомните
имена должны быть осмысленными, пояснять код, вы должны просто прочитать а не хранить в памяти условные обозначения
и даже если вы их помните, но другие разработчики ваших мыслей не читают)

DumDumbIchGB commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -29,40 +29,72 @@
         app:layout_constraintTop_toBottomOf="@+id/numberField">
 
+        <Button
+            android:id="@+id/delete_last_symbol"
```
delete_last_symbol_button - в конце имя типа

DumDumbIchGB commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -29,40 +29,72 @@
+    </LinearLayout>
+
+    <LinearLayout
+        android:id="@+id/firstButtonPanel"
```
стиль кода должен быть snake_case

DumDumbIchGB commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -124,21 +156,21 @@
             android:layout_height="wrap_content"
             android:layout_weight="1"
             android:onClick="onNumberClick"
```
обработчики нажатий нужно определять в коде а не в макете - не нужно смешивать - ухудшается читаемость и поддерживаемость кода

DumDumbIchGB commented on app/src/main/res/layout/activity_second.xml:
```diff
@@ -6,7 +6,7 @@
     android:layout_height="match_parent">
     <TextView
-        android:id="@+id/textView1"
+        android:id="@+id/number_text_view"
```
почти правильно, но не хватает смысла
show_message_text_view - ведь тут можно отображать не только числа - основной смысл - отображение присланного сообщения

DumDumbIchGB commented:
По стилю именования : в XML - snake_case а в коде - camelCase.
У view-элементов - сначала смысловая часть, затем имя типа.

Хорошим тоном считается давать одинаковые имена, различие только в стиле. Например, если в макете определен id  calculation_result_text_view , то в коде соответствующая переменная должна иметь имя  calculationResultTextView - тогда не придется держать в голове, какое имя в коде соответствует определенному id в макете и какой именно элемент скрывается за этим именем.

Не реализован макет для ландшафтной ориентации.

В коде обязательно приведите именования переменных в соотвентствие с принятым стилем.

При повороте экрана появляется лишняя информация

Не реализован класс модели для калькулятора

Stepanav committed:
именования переменных привел в соотвентствие с принятым стилем:
1. исправил на snake_case
2. исправил названия некоторых элементов на более информативные
3. добвил имя типа в конце id

Stepanav committed:
именования переменных привел в соотвентствие с принятым стилем:
1. исправил на snake_case
2. исправил названия некоторых элементов на более информативные
3. добвил имя типа в конце id
4. создал ресурсы ко всему, к чему смог
5. создал горизонтальный экран

Stepanav committed:
Добавил кнопку "Удалить всё"

Stepanav committed:
Добавил кнопку "Удалить последний введенный символ"

Stepanav committed:
Добавил кнопки: "Удалить последний символ" и "Очистить всё"
